kedro-datasets/kedro_datasets_experimental/polars/polars_database_dataset.py

```diff
@@ -0,0 +1,48 @@
+import copy
+from pathlib import PurePosixPath
+from typing import Any, NoReturn
+
+import polars as pl
+
+from kedro_datasets.pandas.sql_dataset import SQLQueryDataset, get_filepath_str
+
+
+class PolarsDatabaseDataset(SQLQueryDataset):
+
+    def __init__(  # noqa: PLR0913
+        self,
+        sql: str | None = None,
+        credentials: dict[str, Any] | None = None,
+        load_args: dict[str, Any] | None = None,
+        fs_args: dict[str, Any] | None = None,
+        filepath: str | None = None,
+        metadata: dict[str, Any] | None = None,
+    ) -> None:
+        """Creates a new ``PolarsDatabaseDataset``."""
+        super().__init__(
+            sql=sql,
+            credentials=credentials,
+            load_args=load_args,
+            fs_args=fs_args,
+            filepath=filepath,
+            metadata=metadata,
+        )
+
+    def load(self) -> pl.DataFrame:
+        load_args = copy.deepcopy(self._load_args)
+
+        if self._filepath:
+            load_path = get_filepath_str(PurePosixPath(self._filepath), self._protocol)
+            with self._fs.open(load_path, mode="r") as fs_file:
+                query = fs_file.read()
+        else:
+            query = load_args.pop("sql")
+
+        return pl.read_database(
+            query=query,
+            connection=self._connection_str,
+            **load_args
```

Comment:
Can you add some docs or checking to reflect this logic
1. If filepath exist use it
2. Otherwise use sql
3. If both are defined, maybe error out or at least log which one is being used?

Ideally I would re-order the argument as well, since the dataset put `sql` as the first argument but actually `filepath` have higher priority which feels counter-intuitive.